publication/models/account_analytic_invoice_line.py

```diff
@@ -28,3 +28,62 @@ def action_distribution_list(self):
         action['view_mode'] = 'form'
         action['target'] = 'current'
         return action
+
+    @api.model
+    def create(self, vals):
+        distribution_list_model = self.env['distribution.list']
+        if vals.get('publication'):
```

Comment:
Wrong test. publication will not be in vals, but is an attribute of the product, and only a reference field on the analytic invoice line. Anyway, computation should only be done after the create, not before.